tensorflow_probability/python/experimental/mcmc/BUILD

```diff
@@ -41,6 +41,7 @@ py_library(
         ":particle_filter_augmentation",
         ":sample_sequential_monte_carlo",
         ":sequential_monte_carlo_kernel",
+        ":streaming_reductions",
```

Comment:
Let's actually put this code in tensorflow_probability/python/experimental/mcmc/sample.py, to imply that we are suggesting it will move to ensorflow_probability/python/mcmc/sample.py when it graduates.  The actual reducers and WithReductions (name TBD) should still go into streaming_reductions as you have here.

Up to you whether you think it will be simpler to make the rearrangement now or in a follow-up PR; also open to you feeling like it's not a good idea at all.

Reply:
Sounds like a good idea. I didn't know if we actually wanted everything driver-like to go into sample.py or a different file. I can make the rearrangement now.

A note on the following reducers: I currently have them set up in different files (i.e. one for the base class, one for covariance reducer, one for WithReductions) to match how tfp.mcmc seems to organize it all right now. I think it's cleaner and better than mushing everything together, but let me know if it's better to group them together in experimental.

Reply:
Separate files are fine.